views/admin/javascripts/exhibits.js

```diff
@@ -468,4 +472,216 @@ Omeka.ExhibitBuilder = {};
+                },
+                error: function(xhr, textStatus, errorThrown) {
+                    alert('Error getting items: ' . textStatus);
+                }
+            });
+        }
+
+        $('#cover-image-container').on('click', '.edit-cover-image', function (event) {
+            var coverImage;
+            event.preventDefault();
+
+            if($(this).prop('id') == 'first-time-cover-image'){
+                coverImagePanel
+                    .removeClass('editing-cover-image')
+                    .removeClass('editing-selection')
+                    .dialog('open');
+            } else {
+                coverImage = $(this).parent();
+                loadItemOptionsForm(getCoverImageData(coverImage));
+                coverImagePanel.addClass('editing-cover-image').dialog('open');
+            }
+        });
+
+        $('#cover-image-container').on('click', '#exhibit-choose-cover-image', function (event) {
+            event.preventDefault();
+            coverImagePanel
+                .removeClass('editing-cover-image')
+                .removeClass('editing-selection')
+                .dialog('open');
+            $('div.item-listing[data-item-id=' + selected_cover_image_id +']').click();
+        });
+
+        $('#item-select').on('click', '.item-listing', function (event) {
+            $('#item-list div.item-selected').removeClass('item-selected');
+            $(this).addClass('item-selected');
+            $('#choose-cover-image').prop('disabled', false);
+        });
+
+        $('#choose-cover-image').on('click', function (event) {
+            file_id = $('input[name="file_id"]:checked').val();
+            event.preventDefault();
+            chooseCoverImage(file_id);
+            coverImagePanel.dialog('close');
+        });
+
+        $('#cover-image-item-options').on('click','.file-select .item-file', function(event) {
+            $(this).find('input[type="radio"]').prop('checked', true);
+            $('.selected').removeClass('selected');
+            $(this).addClass('selected');
+        });
+
+        $('#change-selected-item').on('click', function (event) {
+            event.preventDefault();
+            $('#cover-image-panel')
+                .removeClass('editing-cover-image')
+                .addClass('editing-selection');
+        });
+
+        $('#revert-selected-item').on('click', function (event) {
+            event.preventDefault();
+            $('#cover-image-panel')
+                .removeClass('editing-selection')
+                .addClass('editing-cover-image');
+        });
+    }
+
+    Omeka.ExhibitBuilder.setUpCoverImageSelect = function(browseUri) {
+        /*
+         * Use AJAX to retrieve the list of items that can be attached.
+         */
+        function getItems(uri, parameters) {
+            $('#cover-image-panel').addClass('loading');
+            $.ajax({
+                url: uri,
+                data: parameters,
```

Comment:
It might be a good idea to force `hasImage: 1` into the parameter set here so the selection list only contains items that can be used for thumbnails.